apache/superset
# feat: Add single select and inverse selection to numeric range (#16722)

### SUMMARY
Support the ability to filter by numeric range as a single maximum, minimum, or exact value.

### Filters config form BEFORE:
![Before_1](https://user-images.githubusercontent.com/33639770/140769285-86c4832a-51c7-406b-a9d7-47d5c0fc577e.gif)

### Range filter plugin BEFORE:
![Before_2](https://user-images.githubusercontent.com/33639770/140769404-96749048-9e5f-4416-9b3e-951c4ec83c83.gif)

### Filters config form AFTER:
![After_1](https://user-images.githubusercontent.com/33639770/140769565-738d5061-570f-4788-bda8-5e3b10913248.gif)

### Range filter plugin AFTER, using MAXIMUM value:
![After_2](https://user-images.githubusercontent.com/33639770/140769711-9ea711e3-86c5-483d-8e12-5d528c6604f6.gif)

### Range filter plugin AFTER, using EXACT value:
![After_3](https://user-images.githubusercontent.com/33639770/140769836-7322921c-d948-4b0a-9204-c7106fed7af6.gif)

### Range filter plugin AFTER, using MINIMUM value:
![After_4](https://user-images.githubusercontent.com/33639770/140769904-5697d2ed-5175-4535-98b4-24d850188dd3.gif)

### TESTING INSTRUCTIONS
Using a numerical range filter, choose "Single Value" in the advanced section.  This will reveal 3 single value type options supporting the following behaviors:
	* Minimum: Filter values greater than or equal to the selected value on the range slider.
	* Exact: Filter values equal to the selected value on the range slider.
	* Maximum: Filter values less than or equal to the selected value on the range slider.
	
If "Single Value" is not checked, the previous default range slider behavior applies.
If "Single Value" is checked and Minimum is chosen, the slider should support a single thumb sliding from the right hand side of the control.
If "Single Value" is checked and Exact is chosen, the slider should support a single thumb sliding in the middle of the control with no highlighted range displayed.
If "Single Value" is checked and Maximum is chosen, the slider should support a single thumb sliding from the left hand side of the control.

### ADDITIONAL INFORMATION


- [x] Has associated issue: #16722
